tests/test_cut_diff.py

```diff
@@ -818,16 +932,110 @@ def test_report(self):
+            (
+                "Sequence Cut Summary changes on This is a Test",
+                (
+                    "\n\nLinks: \n\nThe changes in This is a Test are as follows:\n\n"
+                    "28 New Shots\n%s\n\n"
+                    "0 Omitted Shots\n\n\n"
+                    "0 Reinstated Shot\n\n\n"
+                    "0 Cut Changes\n\n\n"
+                    "0 Rescan Needed\n\n\n"
+                ) % "\n".join(["shot_%d" % (6670 + i) for i in range(28)])
+            )
+        )
+        SGSettings().shot_cut_fields_prefix = "myprecious"
+
+        timeline_from_edl = otio.adapters.read_from_file(path)
+        new_track = timeline_from_edl.tracks[0]
+        # Check that using custom Shot cut fields is handled
+        # Validation should fail
+        with self.assertRaisesRegexp(
+                ValueError,
+                "Following SG Shot fields are missing",
+        ):
+            self._get_track_diff(new_track, sg_track, self._mock_compute_clip_shot_name)
+        # Bypass validation and check the fields are passed through
+        with mock.patch.object(SGShotFieldsConfig, "validate_shot_cut_fields_prefix"):
+            track_diff = self._get_track_diff(new_track, sg_track, self._mock_compute_clip_shot_name)
+            self.assertEqual(track_diff.count_for_type(_DIFF_TYPES.NO_CHANGE), 4)
+            self.assertEqual(track_diff.count_for_type(_DIFF_TYPES.NEW), 28)
+
+            # Check the custom fields were queried
+            expected = [x % "myprecious" for x in _ALT_SHOT_FIELDS]
+            for cut_diff in track_diff.diffs_for_type(_DIFF_TYPES.NO_CHANGE):
+                self.assertTrue(
+                    all([x in cut_diff.sg_shot for x in expected])
+                )
+            # Check summary CutDiff iteration and Shot values
+            for i, (shot_name, clip_group) in enumerate(track_diff.items()):
+                self.assertEqual(len(clip_group), 1)
+                cut_diff = clip_group[0]
+                if i < 4:
+                    sg_shot = clip_group.sg_shot
+                    self.assertEqual(sg_shot["id"], self.sg_shots[i]["id"])
+                    self.assertTrue("sg_myprecious_cut_out" in sg_shot)
+                    self.assertTrue("sg_myprecious_status_list" in sg_shot)
+                    self.assertTrue("sg_myprecious_working_duration" in sg_shot)
+                    self.assertTrue("sg_myprecious_head_in" in sg_shot)
+                    self.assertTrue("sg_myprecious_cut_duration" in sg_shot)
+                    self.assertTrue("sg_myprecious_cut_order" in sg_shot)
+                    self.assertTrue("sg_myprecious_tail_out" in sg_shot)
+                    self.assertTrue("sg_myprecious_cut_in" in sg_shot)
+                    self.assertEqual(cut_diff.diff_type, _DIFF_TYPES.NO_CHANGE)
+                    self.assertEqual(clip_group.index, i + 1)
+                    # FIXME: fails with 1001 != 1000 (taken from tk-framework-editorial)
```

Comment:
This fails in the editorial fw as well?

Reply:
right, so we need to get to the bottom of this I'm afraid...

Reply:
yeah, I figured as much, that's why I added the FIXME 😄

Reply:
After searching in the code and debugging meanlessly, I found this:
https://github.com/GPLgithub/tk-framework-editorial/blob/856b3b877dfb5aa63c10ec51c71c496e91222cad/tests/test_cut_diff.py#L422

It's the change of defaults. So we're fine.

Reply:
I added the same settings in the tests in sg-otio and now they pass 😄